core/src/autogluon/core/scheduler/seq_scheduler.py

```diff
@@ -0,0 +1,273 @@
+            result['trial'] = self.trial
+
+            now = time.time()
+            result['time_this_iter'] = now - self.last_reported_time
+            result['time_since_start'] = now - self.trial_started
+            self.last_reported_time = now
+
+            self.training_history[self.trial].append(result)
+
+            if self.trial not in self.config_history:
+                self.config_history[self.trial] = self.task_config
+                if 'util_args' in self.task_config:
+                    self.task_config.pop('util_args')
+
+            self.last_result = result
+        if 'traceback' in result:
+            self.searcher.evaluation_failed(config=self.task_config, **result)
+
+    def terminate(self):
+        pass  # compatibility
+
+
+class LocalSequentialScheduler(object):
+    """ Simple scheduler which schedules all HPO jobs in sequence without any parallelizm
+
+    Parameters
+    ----------
+    train_fn : callable
+        A task launch function for training.
+    resource : dict
+        Computation resources. For example, `{'num_cpus':2, 'num_gpus':1}`
+    searcher : str
+        Searcher (get_config decisions). If str, this is passed to
+        searcher_factory along with search_options.
+    search_options : dict
+        If searcher is str, these arguments are passed to searcher_factory.
+    num_trials : int
+        Maximum number of jobs run in experiment. One of `num_trials`,
+        `time_out` must be given.
+    time_out : float
+        If given, jobs are started only until this time_out (wall clock time).
+        One of `num_trials`, `time_out` must be given.
+    reward_attr : str
+        Name of reward (i.e., metric to maximize) attribute in data obtained
+        from reporter
+    time_attr : str
+        Name of resource (or time) attribute in data obtained from reporter.
+        This attribute is optional for FIFO scheduling, but becomes mandatory
+        in multi-fidelity scheduling (e.g., Hyperband).
+        Note: The type of resource must be int.
+    """
+
+    def __init__(self, train_fn, searcher='auto', **kwargs):
+        self.train_fn = train_fn
+        self.training_history = None
+        self.config_history = None
+        self._reward_attr = kwargs['reward_attr']
+        self.time_attr = kwargs.get('time_attr', None)
+        self.resource = kwargs['resource']
+        self.max_reward = kwargs.get('max_reward')
+        self.searcher: BaseSearcher = self.get_scheduler_(searcher, train_fn, **kwargs)
+        self.init_limits_(kwargs)
+        self.metadata = {
+            'search_space': train_fn.kwspaces,
+            'search_strategy': self.searcher,
+            'stop_criterion': {
+                'time_limits': self.time_out,
+                'max_reward': self.max_reward},
+            'resources_per_trial': self.resource
+        }
+
+        args = kwargs.get('args')
+
+    def init_limits_(self, kwargs):
+        if kwargs.get('num_trials') is None:
```

Comment:
won't this crash if num_trials isn't specified? Have it default to None

Reply:
updated to `if kwargs.get('num_trials', None) is None:`